### ISSUE TITLE
Migration Error

### ISSUE THREAD

**Issue by webrgp** [putyourlightson/craft-blitz#490]
I've stepped into this one by accident: I am working on upgrading a craft site from 3 to 4. Everything went well everything got updated.

While doing some house keeping on the plugins, I've accidentally click to uninstall Blitz and I got:

```
PDOException: SQLSTATE[HY000]: General error: 3730 Cannot drop table 'blitz_elementqueries' referenced by a foreign key constraint 'fk_cfxlcnhhrvopdqxahbogncwvqdbfalnxsecw' on table 'blitz_elementqueryattributes'. in /var/www/html/cms/vendor/yiisoft/yii2/db/Command.php:1302
Stack trace:
#0 /var/www/html/cms/vendor/yiisoft/yii2/db/Command.php(1302): PDOStatement->execute()
#1 /var/www/html/cms/vendor/yiisoft/yii2/db/Command.php(1102): yii\db\Command->internalExecute('DROP TABLE IF E...')
#2 /var/www/html/cms/vendor/craftcms/cms/src/db/Migration.php(335): yii\db\Command->execute()
#3 /var/www/html/cms/vendor/putyourlightson/craft-blitz/src/migrations/Install.php(57): craft\db\Migration->dropTableIfExists('{{%blitz_elemen...')
#4 /var/www/html/cms/vendor/craftcms/cms/src/db/Migration.php(82): putyourlightson\blitz\migrations\Install->safeDown()
#5 /var/www/html/cms/vendor/craftcms/cms/src/db/MigrationManager.php(294): craft\db\Migration->down(true)
#6 /var/www/html/cms/vendor/craftcms/cms/src/base/Plugin.php(165): craft\db\MigrationManager->migrateDown(Object(putyourlightson\blitz\migrations\Install))
#7 /var/www/html/cms/vendor/craftcms/cms/src/services/Plugins.php(620): craft\base\Plugin->uninstall()
#8 /var/www/html/cms/vendor/craftcms/cms/src/controllers/PluginsController.php(79): craft\services\Plugins->uninstallPlugin('blitz')
#9 [internal function]: craft\controllers\PluginsController->actionUninstallPlugin()
#10 /var/www/html/cms/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#11 /var/www/html/cms/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#12 /var/www/html/cms/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('uninstall-plugi...', Array)
#13 /var/www/html/cms/vendor/craftcms/cms/src/web/Application.php(302): yii\base\Module->runAction('plugins/uninsta...', Array)
#14 /var/www/html/cms/vendor/craftcms/cms/src/web/Application.php(627): craft\web\Application->runAction('plugins/uninsta...', Array)
#15 /var/www/html/cms/vendor/craftcms/cms/src/web/Application.php(281): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#16 /var/www/html/cms/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#17 /var/www/html/cms/web/index.php(13): yii\base\Application->run()
#18 {main}

Next yii\db\Exception: SQLSTATE[HY000]: General error: 3730 Cannot drop table 'blitz_elementqueries' referenced by a foreign key constraint 'fk_cfxlcnhhrvopdqxahbogncwvqdbfalnxsecw' on table 'blitz_elementqueryattributes'.
The SQL being executed was: DROP TABLE IF EXISTS `blitz_elementqueries` in /var/www/html/cms/vendor/yiisoft/yii2/db/Schema.php:676
Stack trace:
#0 /var/www/html/cms/vendor/yiisoft/yii2/db/Command.php(1307): yii\db\Schema->convertException(Object(PDOException), 'DROP TABLE IF E...')
#1 /var/www/html/cms/vendor/yiisoft/yii2/db/Command.php(1102): yii\db\Command->internalExecute('DROP TABLE IF E...')
#2 /var/www/html/cms/vendor/craftcms/cms/src/db/Migration.php(335): yii\db\Command->execute()
#3 /var/www/html/cms/vendor/putyourlightson/craft-blitz/src/migrations/Install.php(57): craft\db\Migration->dropTableIfExists('{{%blitz_elemen...')
#4 /var/www/html/cms/vendor/craftcms/cms/src/db/Migration.php(82): putyourlightson\blitz\migrations\Install->safeDown()
#5 /var/www/html/cms/vendor/craftcms/cms/src/db/MigrationManager.php(294): craft\db\Migration->down(true)
#6 /var/www/html/cms/vendor/craftcms/cms/src/base/Plugin.php(165): craft\db\MigrationManager->migrateDown(Object(putyourlightson\blitz\migrations\Install))
#7 /var/www/html/cms/vendor/craftcms/cms/src/services/Plugins.php(620): craft\base\Plugin->uninstall()
#8 /var/www/html/cms/vendor/craftcms/cms/src/controllers/PluginsController.php(79): craft\services\Plugins->uninstallPlugin('blitz')
#9 [internal function]: craft\controllers\PluginsController->actionUninstallPlugin()
#10 /var/www/html/cms/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#11 /var/www/html/cms/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#12 /var/www/html/cms/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('uninstall-plugi...', Array)
#13 /var/www/html/cms/vendor/craftcms/cms/src/web/Application.php(302): yii\base\Module->runAction('plugins/uninsta...', Array)
#14 /var/www/html/cms/vendor/craftcms/cms/src/web/Application.php(627): craft\web\Application->runAction('plugins/uninsta...', Array)
#15 /var/www/html/cms/vendor/craftcms/cms/src/web/Application.php(281): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#16 /var/www/html/cms/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#17 /var/www/html/cms/web/index.php(13): yii\base\Application->run()
#18 {main}

Next craft\errors\MigrationException: An error occurred while executing the "putyourlightson\blitz\migrations\Install migration: SQLSTATE[HY000]: General error: 3730 Cannot drop table 'blitz_elementqueries' referenced by a foreign key constraint 'fk_cfxlcnhhrvopdqxahbogncwvqdbfalnxsecw' on table 'blitz_elementqueryattributes'.
The SQL being executed was: DROP TABLE IF EXISTS `blitz_elementqueries` in /var/www/html/cms/vendor/craftcms/cms/src/db/MigrationManager.php:314
Stack trace:
#0 /var/www/html/cms/vendor/craftcms/cms/src/base/Plugin.php(165): craft\db\MigrationManager->migrateDown(Object(putyourlightson\blitz\migrations\Install))
#1 /var/www/html/cms/vendor/craftcms/cms/src/services/Plugins.php(620): craft\base\Plugin->uninstall()
#2 /var/www/html/cms/vendor/craftcms/cms/src/controllers/PluginsController.php(79): craft\services\Plugins->uninstallPlugin('blitz')
#3 [internal function]: craft\controllers\PluginsController->actionUninstallPlugin()
#4 /var/www/html/cms/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#5 /var/www/html/cms/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#6 /var/www/html/cms/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('uninstall-plugi...', Array)
#7 /var/www/html/cms/vendor/craftcms/cms/src/web/Application.php(302): yii\base\Module->runAction('plugins/uninsta...', Array)
#8 /var/www/html/cms/vendor/craftcms/cms/src/web/Application.php(627): craft\web\Application->runAction('plugins/uninsta...', Array)
#9 /var/www/html/cms/vendor/craftcms/cms/src/web/Application.php(281): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#10 /var/www/html/cms/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#11 /var/www/html/cms/web/index.php(13): yii\base\Application->run()
#12 {main}
```

I hit the back button and everything still in place and working, so not a issue for me, but I figure I report it anyways.

- Craft 4.4.5
- Blitz 4.4.3

**Comment by bencroker** [comment]
Thanks for reporting this! Fixed in https://github.com/putyourlightson/craft-blitz/commit/fe3f97ae67fa982b8b65f486ee6c4d1f79b1d81d and released in 4.4.4. Note that you may now be missing some database tables, even though it appears that nothing changed, therefore I’d suggest you reinstall the plugin, just to be safe.

**Comment by webrgp** [comment]
Will do, thanks!
